framework/src/problems/FEProblemBase.C

```diff
@@ -5002,6 +4992,10 @@ FEProblemBase::solve()
 {
   TIME_SECTION(_solve_timer);
 
+  clearAllDofIndices();
```

Comment:
I think the reasoning is clear, but I think it's still a good idea to put a comment here as to why you ended up needing to do this